File: src/main/java/com/now/nowbot/qq/message/VoiceMessage.java
package com.now.nowbot.qq.message;

public class VoiceMessage extends Message {
    final byte[] data;
    public VoiceMessage(byte[] data) {
        this.data = data;
    }

    public byte[] getData() {
        return data;
    }

    @Override
    public String toString() {
        return "[语音]";
    }
}

File: src/main/java/com/now/nowbot/qq/message/VoiceMessage.kt
package com.now.nowbot.qq.message

import com.now.nowbot.util.QQMsgUtil

class VoiceMessage(val data: ByteArray) : Message() {
    override fun toString(): String {
        return "[语音]"
    }

    override fun toJson(): JsonMessage {
        return JsonMessage("record", mapOf("file" to "base64://${QQMsgUtil.byte2str(data)}"))
    }
}
